### PR TITLE
add support for config subcommand

### PR DESCRIPTION
**Pull request by sidmutha**

### CODE REVIEW

**Comment by evanlucas** [comment]
Thanks for the pull request. Is there any chance you could rebase this off of the master branch?

**Comment by sidmutha** [comment]
Sure.

**Comment by sidmutha** [comment]
I have rebased off master. 
One thing that I can't figure out is a clean way to add descriptions for the subcommands of config. Currently all of them say 'kube config subcommand'. Can you have a look?

File: kubectl.fish [hunk 1]
```diff
@@ -338,3 +353,7 @@ complete -c kubectl -f -n '__fish_kubectl_needs_command' -a version -d 'Print th
 complete -c kubectl -A -f -n '__fish_seen_subcommand_from version' -l client -d 'Client version only (no server required)'
 complete -c kubectl -A -f -n '__fish_seen_subcommand_from version' -s o -l output -a 'yaml json' -d 'Specify output format'
 complete -c kubectl -A -f -n '__fish_seen_subcommand_from version' -l short -a 'true false' -d 'Print just the version number'
+
+# config
+complete -c kubectl -f -n "__fish_kubectl_using_command config; and not __fish_seen_subcommand_from $__kubectl_config_subcommands" -a '$__kubectl_config_subcommands' -d 'kube config subcommand'
+complete -c kubectl -f -n '__fish_kubectl_using_command config; and __fish_seen_subcommand_from use-context delete-context' -a '(kubectl config get-contexts --no-headers | awk \'{print $2}\')'
```
Comment (evanlucas) [inline kubectl.fish]: could you replace `-a '(kubectl config get-contexts --no-headers | awk \'{print $2}\')'` with `-a '(kubectl config get-contexts -o name)'`? It prevents having to call `awk`

**Comment by evanlucas** [comment]
I'm not sure if we will be able to add descriptions for the subcommands of config. I'd be fine with landing this without that though.

**Comment by sidmutha** [comment]
Done.

**Comment by evanlucas** [comment]
Thanks!

**Pull request merged**